pp_sys.c

```diff
@@ -2915,7 +2915,56 @@ PP(pp_stat)
     if (max) {
         EXTEND(SP, max);
         EXTEND_MORTAL(max);
+#if ST_DEV_SIZE < IVSIZE || (ST_DEV_SIZE == IVSIZE && ST_DEV_SIGN < 0)
         mPUSHi(PL_statcache.st_dev);
+#elif ST_DEV_SIZE == IVSIZE
+        mPUSHu(PL_statcache.st_dev);
+#else
+#  if ST_DEV_SIGN < 0
```

Comment:
1. Why isn't ST_DEV_SIGN a boolean?
2. Is a configure probe for st_dev's signedness really needed? The code for `st_ino` manages to do exactly the same thing without it. Come to think of it, I wonder if it would be possible to merge the handling of them somehow...